test/unit/puppetlabs/services/ca/certificate_authority_core_test.clj

```diff
@@ -522,9 +532,51 @@
      (testing "a CSR w/ auth extensions and disallowed auth extensions gets a specific error response"
        (let [csr (io/input-stream (test-pem-file "csr-auth-extension.pem"))
              settings (assoc settings :allow-authorization-extensions false)
-             response (handle-put-certificate-request! "csr-auth-extension" csr settings)]
+             request (assoc request :route-params {:subject "csr-auth-extension"} :body csr)
+             response (handle-put-certificate-request! settings nil request)]
          (is (= 400 (:status response)))
-         (is (re-find #"csr-auth-extension.*disallowed" (:body response))))))))
+         (is (re-find #"csr-auth-extension.*disallowed" (:body response)))))
+         
+     (testing "when authname provided on signing"
+         (let [csr-stream    (io/input-stream static-csr)
+               signee "authname"
+               certname "test-agent"
+               request {:authorization {:name signee} :remote-addr "1.1.1.1" :body csr-stream :route-params {:subject certname}}
+               msg-matcher (re-pattern (str "Entity " signee " signed 1 certificate: " certname))
+               expected-path (ca/path-to-cert (:signeddir settings) certname)]
+            (try
+              (let [response (handle-put-certificate-request! settings nil request)]
+                (is (= 200 (:status response)))
+                (is (logged? msg-matcher :info)))
+            (finally
+             (fs/delete expected-path)))))
+
+     (testing "when rbac-subject provided on signing"
+          (let [csr-stream    (io/input-stream static-csr)
+                signee "rbac-subject"
+                certname "test-agent"
+                request {:rbac-subject {:login signee} :remote-addr "1.1.1.1" :body csr-stream :route-params {:subject certname}}
+                msg-matcher (re-pattern (str "Entity " signee " signed 1 certificate: " certname))
+                expected-path (ca/path-to-cert (:signeddir settings) certname)]
+             (try
+               (let [response (handle-put-certificate-request! settings nil request)]
+                (is (= 200 (:status response))) 
+                (is (logged? msg-matcher :info)))
+             (finally
+               (fs/delete expected-path)))))
+
+     (testing "when no signee info provided"
```

Comment:
You should wrap each of these testing blocks in a `with-test-logging` block, or ideally in their own deftest, to ensure state doesn't get polluted between them.